src/Operators/spectralelement.jl

```diff
@@ -1650,3 +1668,25 @@ function rmatmul2(W, S, i, j)
     end
     return r
 end
+
+
+function apply_operator(
+    op::Union{
+        Divergence{()},
+        WeakDivergence{()},
+        Gradient{()},
+        WeakGradient{()},
+        Curl{()},
+        WeakCurl{()},
+    },
+    space,
+    slabidx,
+    arg,
+)
+    FT = Spaces.undertype(space)
+    RT = operator_return_eltype(op, eltype(arg))
+    zero_value = zero(RT)
+    out = DataLayouts.IF{RT, 1}(MArray, FT)
```

Comment:
Is `IF` always the right layout here? We probably want to use `IJF` when `space` has 2 horizontal dimensions and `DataF` when `space` is a column. You can define some sort of `slab_type` function in `DataLayouts.jl` to abstract this out, if we don't already have something like that.